### PR TITLE
docs: add advanced usage of colcon

### PR DESCRIPTION
**Pull request by kenji-miyake**

### CODE REVIEW

File: docs/how-to-guides/advanced-usage-of-colcon.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+```bash
+VERBOSE=1 colcon build --packages-up-to <package_name> --event-handlers console_cohesion+
+```
+
+See [here](https://colcon.readthedocs.io/en/released/reference/event-handler-arguments.html) for other options.
+
+## Using Ccache
+
+[Ccache](https://ccache.dev/) can speed up recompilation.
+It is recommended to use it to save your time if there are no particular reasons.
```
Comment (xmfcx) [inline docs/how-to-guides/advanced-usage-of-colcon.md:113]:
```suggestion
It is recommended to use it to save your time unless you have a specific reason not to do so.
```

Comment (kenji-miyake) [reply docs/how-to-guides/advanced-usage-of-colcon.md:113]: Thank you!

File: docs/how-to-guides/advanced-usage-of-colcon.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+To see the compiler and linker invocations for a package, use `VERBOSE=1` and `--event-handlers console_cohesion+`:
+
+```bash
+VERBOSE=1 colcon build --packages-up-to <package_name> --event-handlers console_cohesion+
+```
+
+See [here](https://colcon.readthedocs.io/en/released/reference/event-handler-arguments.html) for other options.
+
+## Using Ccache
```
Comment (kenji-miyake) [inline docs/how-to-guides/advanced-usage-of-colcon.md:110]: @xmfcx FYI: Added to the dictionary by this PR.
https://github.com/tier4/autoware-spell-check-dict/pull/224

Comment (kenji-miyake) [reply docs/how-to-guides/advanced-usage-of-colcon.md:110]: And the spell-check CI has passed if I run it again.
https://github.com/autowarefoundation/autoware-documentation/runs/5218266236?check_suite_focus=true

Comment (kenji-miyake) [reply docs/how-to-guides/advanced-usage-of-colcon.md:110]: By the way, if you want to see files without being disturbed by some comments, you can try either:

1. Click the `View file` button
image

2. Use the rich view
image

File: docs/how-to-guides/index.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # How-to guides
 
-!!! warning
+- [Advanced usage of colcon](advanced-usage-of-colcon)
 
-    Under Construction
+TODO: Write the following contents.
 
 - Write an Autoware package
```
Comment (xmfcx) [inline docs/how-to-guides/index.md:7]:
```suggestion
- Create an Autoware package
```

**Review by xmfcx — CHANGES_REQUESTED** [review]

**Review by xmfcx — DISMISSED** [review]

**Comment by kenji-miyake** [comment]
I'll merge this after #18.

**Comment by kenji-miyake** [comment]
Just rebased.

**Pull request merged**